### PR TITLE
refactor(ui,core): remove deprecated code and cleanup

### PR DESCRIPTION
**Pull request by imtoori**

### CODE REVIEW

File: packages/stream_chat_flutter/lib/src/message_input/countdown_button.dart [hunk 1]
```diff
@@ -1,36 +1,32 @@
...
-        decoration: BoxDecoration(
-          color: StreamChatTheme.of(context).colorTheme.disabled,
-          shape: BoxShape.circle,
-        ),
-        child: SizedBox(
-          height: 24,
-          width: 24,
-          child: Center(
-            child: Text('$count'),
+  Widget build(BuildContext context) => Padding(
```
Comment (GroovinChip) [inline packages/stream_chat_flutter/lib/src/message_input/countdown_button.dart:16]: Why is this going back to using the arrow function?

**Review by GroovinChip — CHANGES_REQUESTED** [review]

File: packages/stream_chat_flutter/lib/src/message_input/countdown_button.dart (hunk 1 above)
Comment (imtoori) [reply packages/stream_chat_flutter/lib/src/message_input/countdown_button.dart:16]: I'll revert it, but let's add a linter rule for this if we decided to not use it anymore

Comment (GroovinChip) [reply packages/stream_chat_flutter/lib/src/message_input/countdown_button.dart]: I thought we did that already 🤔

**Review by ayush221b — CHANGES_REQUESTED** [review]

File: packages/stream_chat_flutter/lib/src/message_input/countdown_button.dart [hunk 2]
```diff
@@ -1,17 +1,15 @@
...
+/// Button for showing visual component of slow mode.
+class StreamCountdownButton extends StatelessWidget {
+  /// Constructor for creating [StreamCountdownButton].
+  const StreamCountdownButton({
     super.key,
     required this.count,
   });
 
-  /// The countdown, in seconds.
+  /// Count of time remaining to show to the user.
```
[the remark this answers is not in the document]
Comment (imtoori) [reply packages/stream_chat_flutter/lib/src/message_input/countdown_button.dart:12]:
```suggestion
  /// Count of time remaining to show to the user, in seconds.
```

[the remark this answers is not in the document]
Comment (GroovinChip) [reply packages/stream_chat_flutter/lib/src/message_input/countdown_button.dart]: Maybe rename this to:
`The amount of time remaining until the user can send a message again. Measured in seconds.`

File: packages/stream_chat_flutter/lib/src/message_input/countdown_button.dart [hunk 3]
```diff
@@ -1,17 +1,15 @@
...
+/// Button for showing visual component of slow mode.
+class StreamCountdownButton extends StatelessWidget {
+  /// Constructor for creating [StreamCountdownButton].
+  const StreamCountdownButton({
     super.key,
     required this.count,
   });
 
-  /// The countdown, in seconds.
+  /// Count of time remaining to show to the user, in seconds.
```
Comment (imtoori) [inline packages/stream_chat_flutter/lib/src/message_input/countdown_button.dart:12]:
```suggestion
  /// The amount of time remaining until the user can send a message again. Measured in seconds.
```

**Review by ayush221b — APPROVED** [review]

**Pull request merged**